project-base/storefront/components/Pages/Order/ContactInformation/contactInformationFormMeta.ts

```diff
@@ -193,7 +196,14 @@ export const useContactInformationFormMeta = (
     formProviderMethods: UseFormReturn<ContactInformation>,
 ): ContactInformationFormMetaType => {
     const { t } = useTranslation();
+    const { type } = useDomainConfig();
     const { pickupPlace } = useCurrentCart();
+
+    const user = useCurrentCustomerData();
+    const isUserLoggedIn = user !== undefined;
+    const isCompanyUser = user?.companyCustomer;
+    const isB2B = type === CustomerUserAreaEnum.B2B;
```

Comment:
Why not to use `useCurrentCustomerUserPermissions` and add the new rules there as well?

Reply:
good point... changed the name of that hook as well for better orientation :)